c10/core/SafePyObject.h

```diff
@@ -22,21 +22,33 @@ struct C10_API SafePyObject {
   // Steals a reference to data
   SafePyObject(PyObject* data, c10::impl::PyInterpreter* pyinterpreter)
       : data_(data), pyinterpreter_(pyinterpreter) {}
+  SafePyObject(SafePyObject&& other)
+      : data_(std::exchange(other.data_, nullptr)),
+        pyinterpreter_(other.pyinterpreter_) {}
 
   // In principle this could be copyable if we add an incref to PyInterpreter
   // but for now it's easier to just disallow it.
   SafePyObject(SafePyObject const&) = delete;
   SafePyObject& operator=(SafePyObject const&) = delete;
 
   ~SafePyObject() {
-    (*pyinterpreter_)->decref(data_, /*is_tensor*/ false);
+    if (data_ != nullptr) {
```

Comment:
When would `data_` be a nullptr? why did this change?

Reply:
Oh, because of release below. 

The release does not violate RAII, but the `SafePyObject` is meant to be owning - so it feels odd to have a `release` here while keeping PyObject* data_ a raw ptr. Should we make it a unique pointer?

Reply:
We don't actually own the PyObject.  The Python GC owns the PyObject and has its own reference counting mechanism.  The thing we "own" is remembering to call Py_DECREF() exactly once..

Reply:
We should make that clearer in the doc for this class imo, but understood.